internal/terraform/context_apply2_test.go

```diff
@@ -2814,6 +2814,68 @@ removed {
 	checkStateString(t, state, `<no state>`)
 }
 
+// TestContext2Apply_destroy_and_forget tests that a destroy plan with the forget flag set to true.
+// The expectation is that all resources should be forgotten and not destroyed.
+func TestContext2Apply_destroy_and_forget(t *testing.T) {
```

Comment:
Could you add another test that verifies it still forgets a resource that is in the state but has been removed from the config? That was an edge case that was broken in stacks for general destroy plans and it'd be good to make sure it works here as well.

There's a several iterations we could do for this:

1. test_object.a in state but not in config
2. test_object.a in config with a for each or count, but the for each doesn't evaluate to include test_object.a["orphaned"] or something.
3. Have a deposed resource in state, for a resource that has another instance in configuration and in state.

I can't think of any others, but it'd be good to cover more edge cases to make sure all cases are behaving as expected.